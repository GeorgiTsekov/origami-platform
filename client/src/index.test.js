import React from 'react';
import renderer from 'react-test-renderer';
import Header from './components/header/header';
import TestingEnvironment from './test-utils/router';

describe('Header Component', () => {
    it('shoult render authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: true,
                    id: '123'
                }
            }}>
                <Header />
            </TestingEnvironment>

        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    // it('shout render non-authenticated routes', () => {

    // });
})