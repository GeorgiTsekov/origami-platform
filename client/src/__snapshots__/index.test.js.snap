// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component shoult render authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt=""
    className="logo"
    src="white-origami-bird.png"
  />
  <li
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/"
      onClick={[Function]}
    >
      Publications
    </a>
  </li>
  <li
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/share"
      onClick={[Function]}
    >
      Share your throughts
    </a>
  </li>
  <li
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/profile/123"
      onClick={[Function]}
    >
      Profile
    </a>
  </li>
</header>
`;

exports[`Header Component shout render authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt=""
    className="logo"
    src="white-origami-bird.png"
  />
  <li
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/"
      onClick={[Function]}
    >
      Publications
    </a>
  </li>
  <li
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/share"
      onClick={[Function]}
    >
      Share your throughts
    </a>
  </li>
  <li
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/profile/123"
      onClick={[Function]}
    >
      Profile
    </a>
  </li>
</header>
`;
